test/libs/metrics/top_level_metrics_test.py

```diff
@@ -1,40 +1,112 @@
+        all_columns: All columns to be included in output dataframe.
+        start_date: Optional start date.
+        dates: Optional list of dates to use for rows, length of dates must match
+            length of data in column_data.
+        column_data: Column data with values.  Names of variables should match columns
+            in `all_columns`. All lists must be the same length.
+
+    Returns: DataFrame.
+    """
+    data = {column: column_data.get(column, np.nan) for column in all_columns}
+
+    max_len = max(len(value) for value in data.values() if isinstance(value, list))
+
+    if dates:
+        dates = [pd.Timestamp(date) for date in dates]
+    elif start_date:
+        dates = pd.date_range(start_date, periods=max_len, freq="D")
+    else:
+        raise ValueError("Must set either `start_date` or `dates`")
 
-def _build_metrics_df(content: str) -> pd.DataFrame:
-    header = (
-        "date,fips,caseDensity,testPositivityRatio,contactTracerCapacityRatio,"
-        "infectionRate,infectionRateCI90,icuHeadroomRatio\n"
+    return pd.DataFrame({"date": dates, "fips": fips, **data})
+
+
+def _build_metrics_df(fips, start_date=None, dates=None, **metrics_data) -> pd.DataFrame:
+    """Builds a dataframe that has same structure as those built by
+    `top_level_metrics.calculate_metrics_for_timeseries`
+    """
+    metrics = [
+        "caseDensity",
+        "testPositivityRatio",
+        "contactTracerCapacityRatio",
+        "infectionRate",
+        "infectionRateCI90",
+        "icuHeadroomRatio",
+        "icuCapacityRatio",
+    ]
+    return _build_timeseries_dataframe(
+        fips, metrics, start_date=start_date, dates=dates, **metrics_data
     )
-    data = io.StringIO(f"{header}\n{content}")
-    return common_df.read_csv(data, set_index=False)
 
 
 def _series_with_date_index(data, date: str = "2020-08-25", **series_kwargs):
     date_series = pd.date_range(date, periods=len(data), freq="D")
     return pd.Series(data, index=date_series, **series_kwargs)
 
 
-def _fips_csv_to_one_region(csv_str: str, region: Region) -> OneRegionTimeseriesDataset:
+def _fips_csv_to_one_region(
+    csv_str: str, region: Region, latest=None
+) -> OneRegionTimeseriesDataset:
     df = read_csv_and_index_fips_date(csv_str).reset_index()
     # from_timeseries_and_latest adds the location_id column needed by get_one_region
-    return MultiRegionDataset.from_fips_timeseries_df(df).get_one_region(region)
+    dataset = MultiRegionDataset.from_fips_timeseries_df(df).get_one_region(region)
+    if latest:
+        return dataclasses.replace(dataset, latest=latest)
+    else:
+        return dataset
+
+
+def _fips_df_to_one_region(df, region: Region, latest=None) -> OneRegionTimeseriesDataset:
+    # from_timeseries_and_latest adds the location_id column needed by get_one_region
+    dataset = MultiRegionDataset.from_fips_timeseries_df(df).get_one_region(region)
+    if latest:
+        return dataclasses.replace(dataset, latest=latest)
```

Comment:
I'm fairly wary of using replace with OneRegionTimeseriesDataset.latest. Unlike MultiRegionDataset it doesn't (yet) have `__post_init__` checks that the data has the intended structure. Some subset of the latest values are derived from the timeseries data in OneRegionTimeseriesDataset.data so _fips_df_to_one_region could quite easily return an object that won't exist outside of a test.
Long term I think it is worth giving OneRegionTimeseriesDataset the same treatment MultiRegionDataset got where static and timeseries are separated and `latest` is derived from them dynamically.
I think tests will be easier to read and maintain if the parameter used to tweak `latest` only modified the dict that is already in the OneRegionTimeseriesDataset.